Updating Kotlin/JS docs for Kotlin 2.2.20 release (#5051)
JetBrains/kotlin-web-site: docs/topics/js/js-to-kotlin-interop.md

It's also possible to apply the `@JsStatic` annotation to a property of an object or a companion object, making its getter
and setter methods static members in that object or the class containing the companion object.

> ### Usage of `BigInt` type to represent Kotlin's `Long` type

Reviewer: ```suggestion
### Use `BigInt` type to represent Kotlin's `Long` type
```

<primary-label ref="experimental-general"/>

> Kotlin/JS uses JavaScript's built-in `BigInt` type to represent Kotlin's `Long` values when compiling to modern JavaScript (ES2020).

Reviewer: ```suggestion
Kotlin/JS uses JavaScript's built-in `BigInt` type to represent Kotlin `Long` values when compiling to modern JavaScript (ES2020).
```

> To enable the usage of `BigInt` type, you need to add the following compiler option to your

Reviewer: ```suggestion
To enable support for the `BigInt` type, you need to add the following compiler option to your
```
This sounds more natural to me. What do you think?

`build.gradle(.kts)` file:

```kotlin
// build.gradle.kts
kotlin {
    js {
        ...
        compilerOptions {
            freeCompilerArgs.add("-Xes-long-as-bigint")
        }
    }
}
```

> This feature is [Experimental](components-stability.md#stability-levels-explained). Please
> report any problems in our issue tracker, [YouTrack](https://youtrack.jetbrains.com/issue/KT-57128/KJS-Use-BigInt-to-represent-Long-values-in-ES6-mode).

Reviewer: ```suggestion
This feature is [Experimental](components-stability.md#stability-levels-explained). Share you feedback in our [issue tracker](https://youtrack.jetbrains.com/issue/KT-57128/KJS-Use-BigInt-to-represent-Long-values-in-ES6-mode).
```
Removing "please" as we don't need to be so polite in our documentation :) I also suggested a more neutral phrase when asking for feedback.

> #### Usage of `Long` in exported declarations

Reviewer: ```suggestion
#### Use `Long` in exported declarations
```

> Given that Kotlin's `Long` type can be compiled to JavaScript's `BigInt` type, Kotlin/JS supports exporting `Long` values to JavaScript.

Reviewer: ```suggestion
Since Kotlin's `Long` type can compile to JavaScript's `BigInt` type, Kotlin/JS supports exporting `Long` values to JavaScript.
```

To enable this feature:

> 1. Allow exporting `Long` in Kotlin/JS. Add the following compiler argument to the `freeCompilerArgs` attribute

Reviewer: ```suggestion
1. Allow exporting `Long` in Kotlin/JS. Add the following compiler option to the `freeCompilerArgs` attribute
```
We say "compiler option" instead of "argument".

   in your `build.gradle(.kts)` file:

 ```kotlin
// build.gradle.kts
kotlin {
    js {
        ...
        compilerOptions {                   

>        freeCompilerArgs.add("-XXLanguage:+JsAllowLongInExportedDeclarations")

Reviewer: ```suggestion
            freeCompilerArgs.add("-XXLanguage:+JsAllowLongInExportedDeclarations")
```

        }
    }
}
```

> 2. Enable the `BigInt` type. See how to enable it in [Usage of `BigInt` type to represent Kotlin's `Long` type](#usage-of-bigint-type-to-represent-kotlin-s-long-type).

Reviewer: ```suggestion
2. Enable the `BigInt` type. See how to enable it in [Use`BigInt` type to represent Kotlin's `Long` type](#use-bigint-type-to-represent-kotlin-s-long-type).
```

## Kotlin types in JavaScript

See how Kotlin types are mapped to JavaScript ones:

| Kotlin                                                           | JavaScript                | Comments                                                                                |
|------------------------------------------------------------------|---------------------------|-----------------------------------------------------------------------------------------|
| `Byte`, `Short`, `Int`, `Float`, `Double`                        | `Number`                  |                                                                                         |
| `Char`                                                           | `Number`                  | The number represents the character's code.                                             |

> | `Long`                                                           | `BigInt`                  | Needs the `-Xes-long-as-bigint` compiler option configured.                             |

Reviewer: Can you link to the compiler reference here? :)

> | `Long`                                                           | `BigInt`                  |                                                                                         |

Reviewer 2: Could we highlight in some way that it works only if `bigint` is used?

> | `Long`                                                           | `BigInt`                  | `Long` only works when `BigInt` is used.                                                |

Reviewer 2: ```suggestion
| `Long`                                                           | `BigInt`                  | It works only with the `-Xes-long-as-bigint` flag is turned on.                                                |
```

> | `Long`                                                           | `BigInt`                  | It works only with the `-Xes-long-as-bigint` flag turned on.                                                |

Reviewer: ```suggestion
| `Long`                                                           | `BigInt`                  | Needs the `-Xes-long-as-bigint` compiler option to be configured.                                                |
```
"Compiler option" rather than "flag".
